staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -66,6 +66,24 @@ type CompilationResult struct {
 	MaxCost uint64
 }
 
+var baseEnv *cel.Env
+
+func init() {
+	var err error
+	var opts []cel.EnvOption
+	opts = append(opts, cel.HomogeneousAggregateLiterals())
+	opts = append(opts, cel.StdLib())
+	opts = append(opts, library.ExtensionLibs...)
+	// Validate function declarations once during base env initialization,
+	// so they don't need to be evaluated each time a CEL rule is compiled.
+	// This is a relatively expensive operation.
+	opts = append(opts, cel.EagerlyValidateDeclarations(true))
+	baseEnv, err = cel.NewCustomEnv(opts...)
```

Comment:
You can actually write this the following way and it will just work:

```go
var opts []cel.EnvOption
opts = append(opts, cel.HomogeneousAggregateLiterals())
opts = append(opts, cel.EagerlyValidateDeclarations(true))
opts = append(opts, library.ExtensionLibs...)
baseEnv, err = cel.NewEnv(opts...)
```

The standard environment is pre-initialized within cel-go, so calls to `NewEnv()` will actually extend a base standard environment just like you're doing here.

Reply:
Thx, updated.